src/grammar.coffee

```diff
@@ -135,36 +135,35 @@ grammar =
 
   # A literal identifier, a variable name or property.
   Identifier: [
-    o 'IDENTIFIER',                             -> new Literal $1
+    o 'IDENTIFIER',                             -> new IdentifierLiteral $1
   ]
 
   # Alphanumerics are separated from the other **Literal** matchers because
   # they can also serve as keys in object literals.
   AlphaNumeric: [
-    o 'NUMBER',                                 -> new Literal $1
+    o 'NUMBER',                                 -> new NumberLiteral $1
```

Comment:
@lydell I'm not talking about a reference to the `Infinity` global, I'm talking about infinity values (such as the initial value of the `Infinity` global or the value created by the literals `2e308` or 500 `9`s in a row).